apps/courses/cms_wizards.py

```diff
@@ -0,0 +1,198 @@
+"""
+CMS Wizard to add a course page
+"""
+from django import forms
+from django.template.defaultfilters import slugify
+from django.utils.translation import get_language, ugettext_lazy as _
+
+from cms.api import create_page
+from cms.forms.wizards import SlugWidget
+from cms.models import Page
+from cms.wizards.wizard_base import Wizard
+from cms.wizards.wizard_pool import wizard_pool
+
+from apps.organizations.models import Organization
+
+from .models import Course, Subject
+
+
+class BaseWizardForm(forms.Form):
+    """
+    Factorize the part of the wizard form that handles the title and the slug as they are
+    common to all our wizard forms.
+    """
+    title = forms.CharField(
+        max_length=255,
+        required=True,
+        widget=forms.TextInput(),
+        label=_("Page title"),
+        help_text=_("Title of the page in current language"),
+    )
+    slug = forms.CharField(
+        max_length=200,  # Should be less than 255, the "max_length" of a Page's "path" field
+        required=False,
+        widget=SlugWidget(),
+        label=_("Page slug"),
+        help_text=_("Slug of the page in current language"),
+    )
+
+    def clean(self):
+        """
+        Ensure that the slug field is set or derive it from the title
+        """
+        cleaned_data = super().clean()
+        # If the slug is not explicitly set, generate it from the title
+        if cleaned_data.get("title") and not cleaned_data.get("slug"):
+            cleaned_data["slug"] = slugify(cleaned_data["title"])[:200]
+
+        return cleaned_data
+
+    def clean_slug(self):
+        """
+        First check that a parent page exists under which we can create the new page,
+        then check that the length of the slug is compatible with this parent page:
+
+        A page `path` is limited to 255 chars, therefore the course page slug should
+        always be shorter than (255 - length of parent page path - 1 character for the "/")
+        """
+        parent_page = Page.objects.filter(
+            reverse_id=self.model.ROOT_REVERSE_ID, publisher_is_draft=True
+        ).first()
+        if not parent_page:
+            raise forms.ValidationError(
+                _(
+                    "You must first create a parent page and set its `reverse_id` to "
```

Comment:
This error pops up when attempting to create a new subject or organization.

I'm not really sure what's expected of me at that point...

- Either we should be able to create a Subject / Organization from the wizard
- Or the wizard should not even show the "Organization" & "Subject" options?

Reply:
This is because we create the course pages as children of a "all courses" page. So we need it to exist and identify it by its reverse_id set to "courses" (see page advanced settings).

You can create a basic site with all the necessary root pages by running the command "create_cms_data".

I need to add some documentation.

Reply:
That was it!